app/Makefile

```diff
@@ -131,7 +131,20 @@ endif
 ############
 # Platform
 
-DEFINES       += PRINTF\(...\)=
+ifndef DEBUG
+	DEBUG = 0
+endif
+
+ifneq ($(DEBUG),0)
+	DEFINES += HAVE_PRINTF
+	ifeq ($(TARGET_NAME),TARGET_NANOS)
+    	DEFINES += PRINTF=screen_printf
+	else
+    	DEFINES += PRINTF=mcu_usb_printf
+    endif
+else
+    DEFINES += PRINTF\(...\)=
```

Comment:
Quick note:
Be careful when using the PRINTF macro as its current behavior is a bit dangerous
example:
`PRINTF("%u\n", ++var);`
will change code depending on the DEBUG flag
(PR is open on the SDK to rationalize it)

Reply:
Ok thanks for the tip. I'll be careful.
Do you have a link to the PR ?

Reply:
No link yet as the SDK will move to another host soon

Reply:
Ok.